src/Opdex.Platform.Application/EntryHandlers/Tokens/CreateTokenCommandHandler.cs

```diff
@@ -14,40 +17,57 @@ namespace Opdex.Platform.Application.EntryHandlers.Tokens;
     }
 
     public async Task<ulong> Handle(CreateTokenCommand request, CancellationToken cancellationToken)
     {
+        var attributes = request.Attributes;
         var token = await _mediator.Send(new RetrieveTokenByAddressQuery(request.Token, findOrThrow: false), CancellationToken.None);
         var tokenId = token?.Id ?? 0ul;
 
-        if (token == null)
+        if (token is not null)
         {
-            var summary = await _mediator.Send(new CallCirrusGetStandardTokenContractSummaryQuery(request.Token,
-                                                                                                  request.BlockHeight,
-                                                                                                  includeBaseProperties: true,
-                                                                                                  includeTotalSupply: true));
-
-            token = new Token(request.Token,
-                              summary.Name,
-                              summary.Symbol,
-                              (int)summary.Decimals.GetValueOrDefault(),
-                              summary.Sats.GetValueOrDefault(),
-                              summary.TotalSupply.GetValueOrDefault(),
-                              request.BlockHeight);
-
-            tokenId = await _mediator.Send(new MakeTokenCommand(token, request.BlockHeight));
+            return tokenId;
         }
 
-        await Task.WhenAll(request.Attributes.Select(attribute =>
+        var summary = await _mediator.Send(new CallCirrusGetStandardTokenContractSummaryQuery(request.Token,
+            request.BlockHeight,
+            includeBaseProperties: true,
+            includeTotalSupply: true));
+
+        token = new Token(request.Token,
+            summary.Name,
+            summary.Symbol,
+            (int)summary.Decimals.GetValueOrDefault(),
+            summary.Sats.GetValueOrDefault(),
+            summary.TotalSupply.GetValueOrDefault(),
+            request.BlockHeight);
+
+        tokenId = await _mediator.Send(new MakeTokenCommand(token, request.BlockHeight));
+        if (tokenId == 0) _logger.LogError("Something went wrong indexing the token");
+
+        var interfluxSummary = await _mediator.Send(new CallCirrusGetInterfluxTokenContractSummaryQuery(request.Token, request.BlockHeight), CancellationToken.None);
+        if (interfluxSummary is not null)
+        {
+            attributes = request.Attributes.Append(TokenAttributeType.Interflux);
+
+            var tokenWrapped = new TokenWrapped(tokenId, interfluxSummary.Owner, interfluxSummary.NativeChain, interfluxSummary.NativeAddress, request.BlockHeight);
+            var tokenWrappedId = await _mediator.Send(new MakeTokenWrappedCommand(tokenWrapped), CancellationToken.None);
+            if (tokenWrappedId == 0) _logger.LogError("Something went wrong indexing the wrapped token mapping");
+        }
```

Comment:
I think it might be worth making a CreateTokenWrappedCommand here and a `CreateTokenAttributesCommand` for below to share between the create token command handlers.  Not exactly pressing for it, but it would be nicer to maintain.

Reply:
I'm not sure this would make maintenance any easier tbh. Domain events might be nice to implement and would ease maintenance a lot, but that's a big time investment.